snapcraft/_store.py

```diff
@@ -272,12 +274,99 @@ def download(snap_name, channel, download_path, arch):
     try:
         with _requires_login():
             store.download(snap_name, channel, download_path, arch)
-    except storeapi.errors.SnapNotFoundError:
-        raise RuntimeError(
-            'Snap {name} for {arch} cannot be found'
-            ' in the {channel} channel'.format(name=snap_name, arch=arch,
-                                               channel=channel))
     except storeapi.errors.SHAMismatchError:
         raise RuntimeError(
             'Failed to download {} at {} (mismatched SHA)'.format(
                 snap_name, download_path))
+
+
+def gated(snap_name):
+    """Print list of snaps gated by snap_name."""
+    store = storeapi.StoreClient()
+    # Get data for the gating snap
+    with _requires_login():
+        snap_data = store.cpi.get_package(snap_name, 'stable', None)
+    snap_id = snap_data['snap_id']
+    validations = store.get_validations(snap_id)
+
+    table_data = []
+    for v in validations:
+        table_data.append([v['approved-snap-name'],
+                           v['approved-snap-revision']])
+    tabulated = tabulate(table_data, headers=['Name', 'Approved'],
+                         tablefmt="plain")
+    print(tabulated)
+
+
+def validate(snap_name, validations, revoke=False, key=None):
+    """Generate, sign and upload validation assertions."""
+
+    # Check validations format
+    _check_validations(validations)
+
+    store = storeapi.StoreClient()
+
+    # Get data for the gating snap
+    with _requires_login():
+        snap_data = store.cpi.get_package(snap_name, 'stable', None)
+    release = str(snap_data['release'][0])
+    snap_id = snap_data['snap_id']
+
+    # Need the ID of the logged in user.
+    with _requires_login():
+        account_info = store.get_account_information()
+    authority_id = account_info['account_id']
+
+    # Then, for each requested validation, generate assertion
+    for validation in validations:
+        gated_name, rev = validation.split('=', 1)
+        approved_data = store.cpi.get_package(gated_name, 'stable', None)
```

Comment:
Right, for the validated snaps (3rd-part most likely) we hit CPI and require stable publications __ for now, because I wonder if we would even set things up while all snaps are in unstable channels, but I am sure we can sort this out later.
None is not needed.